### PR TITLE
Add optional traits::make and use it in cross_product() and make().

### PR DESCRIPTION
**Pull request by awulkiew**
This change addresses the backward compatibility issue raised here: https://github.com/boostorg/geometry/pull/745

With the new optional trait it is possible to define how a geometry like Point, Box or Segment is constructed.

`bg::make()` for 2d and 3d points is `constexpr` now if the trait is specialized.

### CODE REVIEW

File: include/boost/geometry/geometries/point_xyz.hpp [hunk 1]
```diff
@@ -124,6 +124,21 @@ struct access<model::d3::point_xyz<CoordinateType, CoordinateSystem>, Dimension>
...
 };
 
+template<typename CoordinateType, typename CoordinateSystem>
+struct make<model::d3::point_xyz<CoordinateType, CoordinateSystem> >
+{
+    typedef model::d3::point_xyz<CoordinateType, CoordinateSystem> point_type;
+
+    static const bool is_specialized = true;
+
+    static constexpr point_type apply(CoordinateType const& v0,
```
Comment (vissarion) [inline include/boost/geometry/geometries/point_xyz.hpp:134]: do you need 3 coordinates here?

File: include/boost/geometry/geometries/point_xyz.hpp [hunk 2]
```diff
@@ -124,6 +124,21 @@ struct access<model::d3::point_xyz<CoordinateType, CoordinateSystem>, Dimension>
...
+        CoordinateType const& v1)
```
Comment (vissarion) [inline include/boost/geometry/geometries/point_xyz.hpp:135]: indentation issue

**Review by vissarion — APPROVED** [review]
Looks OK to me. Thanks!
As an addition, it could be useful to add some basic unit tests.

File: include/boost/geometry/geometries/point_xyz.hpp (hunk 1 above)
Comment (awulkiew) [reply include/boost/geometry/geometries/point_xyz.hpp:134]: Yes, I'll fix it.

**Comment by awulkiew** [comment]
> it could be useful to add some basic unit tests.

Right.

**Comment by awulkiew** [comment]
What do you think about moving `point_xy` and `point_xyz` from `model::d2` and `model::d3` namespace into `model` namespace? Of course leaving the alias for backward compatibility for now. These additional namespaces seems to be not needed. Or is there a reason why they are there?

**Comment by vissarion** [comment]
> What do you think about moving `point_xy` and `point_xyz` from `model::d2` and `model::d3` namespace into `model` namespace? Of course leaving the alias for backward compatibility for now. These additional namespaces seems to be not needed. 

I am ok with that.

>Or is there a reason why they are there?

@barendgehrels should know

File: include/boost/geometry/geometries/point_xyz.hpp (hunk 2 above)
Comment (awulkiew) [reply include/boost/geometry/geometries/point_xyz.hpp]: done

**Comment by awulkiew** [comment]
@barendgehrels are you ok with this change?

**Comment by barendgehrels** [comment]
> 
> 
> > What do you think about moving `point_xy` and `point_xyz` from `model::d2` and `model::d3` namespace into `model` namespace? Of course leaving the alias for backward compatibility for now. These additional namespaces seems to be not needed.
> 
> I am ok with that.
> 
> > Or is there a reason why they are there?
> 
> @barendgehrels should know

I'm OK with the move. But it is not backwards compatible, users might use it. We should add it into the release notes then. And not for 1.75 then...

**Review by barendgehrels — APPROVED** [review]
I'm OK, sorry for the delay.

**Comment by awulkiew** [comment]
@barendgehrels Thanks!

> I'm OK with the move. But it is not backwards compatible, users might use it. We should add it into the release notes then. And not for 1.75 then...

Yes, it's too late for that. I think we should add aliases in `d2` and `d3` namespaces and this way this change would be backward compatible.

**Pull request merged**